packages/odyssey-react-mui/src/labs/Switch.tsx

```diff
@@ -0,0 +1,238 @@
+  checked: boolean;
+  hint: SwitchProps["hint"];
+  hintId?: string;
+  isFullWidth: SwitchProps["isFullWidth"];
+  label: SwitchProps["label"];
+};
+
+const SwitchLabel = ({
+  checked,
+  hint,
+  hintId,
+  isFullWidth,
+  label,
+}: SwitchLabelProps) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  return (
+    <>
+      <Box
+        sx={{
+          display: "flex",
+          alignItems: "center",
+          flexWrap: "wrap",
+          gap: odysseyDesignTokens.Spacing1,
+          margin: 0,
+          maxWidth: isFullWidth
+            ? "100%"
+            : odysseyDesignTokens.TypographyLineLengthMax,
+          fontWeight: odysseyDesignTokens.TypographyWeightBodyBold,
+        }}
+      >
+        {label}
+        <Box
+          sx={{
+            padding: "2px 4px",
+            backgroundColor: checked
+              ? odysseyDesignTokens.PaletteSuccessLighter
+              : odysseyDesignTokens.HueNeutral100,
+            borderRadius: odysseyDesignTokens.BorderRadiusMain,
+            color: checked
+              ? odysseyDesignTokens.PaletteSuccessText
+              : odysseyDesignTokens.HueNeutral700,
+            fontWeight: odysseyDesignTokens.TypographyWeightBodyBold,
+            fontSize: odysseyDesignTokens.TypographyScale0,
+            lineHeight: odysseyDesignTokens.TypographyLineHeightOverline,
+            transitionProperty: "background-color, color",
+            transitionDuration: odysseyDesignTokens.TransitionDurationMain,
+          }}
+        >
+          {checked ? "Active" : "Inactive"}
+        </Box>
+      </Box>
+      {hint && <FieldHint id={hintId} text={hint} />}
+    </>
+  );
+};
+
+const Switch = ({
+  hint,
+  id: _id,
+  isChecked,
+  isDefaultChecked,
+  isDisabled,
+  isFullWidth = false,
+  label,
+  name: _name,
+  onChange,
+  testId,
+  value = "Something",
+}: SwitchProps) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+  const controlledStateRef = useRef(
+    getControlState({
+      controlledValue: isChecked,
+      uncontrolledValue: isDefaultChecked,
+    })
+  );
+  const inputValues = useMemo(() => {
+    if (controlledStateRef.current === ComponentControlledState.CONTROLLED) {
+      return { checked: isChecked };
+    }
+    return { defaultChecked: isDefaultChecked };
+  }, [isDefaultChecked, isChecked]);
+
+  const [checked, setChecked] = useState(
+    Boolean(inputValues.checked || inputValues.defaultChecked)
+  );
+
+  useEffect(() => setChecked(Boolean(isChecked)), [isChecked]);
+
+  const id = useUniqueId(_id);
+
+  const hintId = hint ? `${id}-hint` : undefined;
+  const labelElementId = `${id}-label`;
+
+  const handleOnChange = useCallback(
+    (_: SyntheticEvent<Element, Event>, checked: boolean) => {
+      setChecked(checked);
+      onChange?.({ checked, value });
+    },
+    [onChange, value]
+  );
+
+  const renderSwitchComponent = useMemo(
+    () => (
+      <MuiSwitch
+        checked={checked}
```

Comment:
This needs to be removed and replaced instead with `{ ...inputValues }`. This will either set `checked` or `defaultChecked` props on `MuiSwitch` for you.

Reply:
Chatted with Chris about this on Zoom and resolved